drivers/flash/flash_ifx_cat1_qspi.c

```diff
@@ -0,0 +1,339 @@
+	.addrWidth = CY_SMIF_WIDTH_SINGLE,
+	/* The 8-bit mode byte. This value is 0xFFFFFFFF when there is no mode present. */
+	.mode = 0xFFFFFFFFU,
+	/* The width of the mode command transfer. */
+	.modeWidth = CY_SMIF_WIDTH_SINGLE,
+	/* The number of dummy cycles. A zero value suggests no dummy cycles. */
+	.dummyCycles = 8U,
+	/* The width of the data transfer. */
+	.dataWidth = CY_SMIF_WIDTH_SINGLE,
+};
+
+cy_stc_smif_octal_ddr_en_seq_t oe_sequence_SFDP_SlaveSlot_0 = {
+	.cmdSeq1Len = CY_SMIF_SFDP_ODDR_CMD_SEQ_MAX_LEN,
+	.cmdSeq2Len = CY_SMIF_SFDP_ODDR_CMD_SEQ_MAX_LEN,
+	.cmdSeq1 = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
+	.cmdSeq2 = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
+};
+
+#if (CY_SMIF_DRV_VERSION_MAJOR > 1) || (CY_SMIF_DRV_VERSION_MINOR >= 50)
+static cy_stc_smif_hybrid_region_info_t sfdp_slave_slot_0_region_info_storage[16];
+
+#define GENERATE_REGION_INFO_PTR(index, _) &sfdp_slave_slot_0_region_info_storage[index],
+
+static cy_stc_smif_hybrid_region_info_t *sfdp_slave_slot_0_region_info[16] = {
+	LISTIFY(16, GENERATE_REGION_INFO_PTR, ())};
+#endif
+
+cy_stc_smif_mem_device_cfg_t deviceCfg_SFDP_SlaveSlot_0 = {
+	/* Specifies the number of address bytes used by the memory slave device. */
+	.numOfAddrBytes = 0x03U,
+	/* The size of the memory. */
+	.memSize = 0x0000100U,
+	/* Specifies the Read command. */
+	.readCmd = &sfdp_slave_slot_0_read_cmd,
+	/* Specifies the Write Enable command. */
+	.writeEnCmd = &sfdp_slave_slot_0_write_en_cmd,
+	/* Specifies the Write Disable command. */
+	.writeDisCmd = &sfdp_slave_slot_0_write_dis_cmd,
+	/* Specifies the Erase command. */
+	.eraseCmd = &sfdp_slave_slot_0_erase_cmd,
+	/* Specifies the sector size of each erase. */
+	.eraseSize = 0x0001000U,
+	/* Specifies the Chip Erase command. */
+	.chipEraseCmd = &sfdp_slave_slot_0_chip_erase_cmd,
+	/* Specifies the Program command. */
+	.programCmd = &sfdp_slave_slot_0_program_cmd,
+	/* Specifies the page size for programming. */
+	.programSize = 0x0000100U,
+	/* Specifies the command to read the QE-containing status register. */
+	.readStsRegQeCmd = &sfdp_slave_slot_0_read_sts_reg_qe_cmd,
+	/* Specifies the command to read the WIP-containing status register. */
+	.readStsRegWipCmd = &sfdp_slave_slot_0_read_sts_reg_wip_cmd,
+	/* Specifies the read SFDP command */
+	.readSfdpCmd = &sfdp_slave_slot_0_read_sfdp_cmd,
+	/* Specifies the command to write into the QE-containing status register. */
+	.writeStsRegQeCmd = &sfdp_slave_slot_0_write_sts_reg_qe_cmd,
+	/* The mask for the status register. */
+	.stsRegBusyMask = 0x00U,
+	/* The mask for the status register. */
+	.stsRegQuadEnableMask = 0x00U,
+	/* The max time for the erase type-1 cycle-time in ms. */
+	.eraseTime = 1U,
+	/* The max time for the chip-erase cycle-time in ms. */
+	.chipEraseTime = 16U,
+	/* The max time for the page-program cycle-time in us. */
+	.programTime = 8U,
+#if (CY_SMIF_DRV_VERSION_MAJOR > 1) || (CY_SMIF_DRV_VERSION_MINOR >= 50)
```

Comment:
Same version check as above, so if that logic had to change, change this logic as well..

Reply:
updated the logic.